File: Python/1720.py
# 1720. Decode XORed Array
# https://leetcode.com/problems/decode-xored-array/

class Solution:
  def decode(self, encoded: List[int], first: int) -> List[int]:
    out = [first]

    for i in range(0, len(encoded)):
      out.append(encoded[i] ^ out[i])

    return out

File: Rust/1720.rs
/**
 * 1720. Decode XORed Array
 * https://leetcode.com/problems/decode-xored-array/
 */

impl Solution {
  pub fn decode(encoded: Vec<i32>, first: i32) -> Vec<i32> {
    let mut out: Vec<i32> = vec![first];

    for i in 0..encoded.len() {
      out.push(encoded[i] ^ out[i]);
    }

    return out;
  }
}
